src/backends/backend.js

```diff
@@ -234,7 +233,20 @@ class Backend {
     .then(this.entryWithFormat(collection, slug));
   }
 
-  persistEntry(config, collection, entryDraft, MediaFiles, integrations, options = {}) {
+  async checkOverwrite(collection, slug, path, entryDraft) {
+    const errorMessage = "Oops, duplicate filename found. Please choose a unique name."
+    const existingEntry = window.repoFilesUnpublished.find(e => {
```

Comment:
Unfortunately, `repoFilesUnpublished` is an implementation detail of the test-repo backend, not something most backend use, so this will have to be checked using a different method.

Reply:
To get them for any backend type, you should be able to use `backend.unpublishedEntries()`.

Reply:
It may be easier and more performant, however, to just check for the specific entry that you want using `unpublishedEntry()`, instead of filtering them yourself.

Reply:
@tech4him1 the problem is when `backend.unpublishedEntry` uses `implementation.unpublishedEntry` it throws an error if there is no entry found.

```
unpublishedEntry(collection, slug) {
    const entry = window.repoFilesUnpublished.find(e => (
      e.metaData.collection === collection.get('name') && e.slug === slug
    ));
    if (!entry) {
      return Promise.reject(new EditorialWorkflowError('content is not under editorial workflow', true));
    }
    return Promise.resolve(entry);
  }
```

<img width="661" alt="screen shot 2018-04-09 at 3 59 43 pm" src="https://user-images.githubusercontent.com/23248886/38519846-11d6dd14-3c0f-11e8-971e-f89bb71b2157.png">
Ideas to get around this?

Reply:
@tech4him1 badass!